Kentico.Kontent.Management/Modules/ModelBuilders/ElementModelProvider.cs

```diff
@@ -0,0 +1,161 @@
+                    Value = (source?.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id))),
+                };
+            }
+            else if (type == typeof(DateTimeElement))
+            {
+                return new DateTimeElement
+                {
+                    Element = Reference.ById(Guid.Parse(source?.element?.id)),
+                    Value = Convert.ToDateTime(source?.value)
+                };
+            }
+            else if (type == typeof(LinkedItemsElement))
+            {
+                return new LinkedItemsElement
+                {
+                    Element = Reference.ById(Guid.Parse(source?.element?.id)),
+                    Value = (source?.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
+                };
+            }
+            else if (type == typeof(MultipleChoiceElement))
+            {
+                return new MultipleChoiceElement
+                {
+                    Element = Reference.ById(Guid.Parse(source?.element?.id)),
+                    Value = (source?.value as IEnumerable<dynamic>).Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
+                };
+            }
+            else if (type == typeof(TaxonomyElement))
+            {
+                return new TaxonomyElement
+                {
+                    Element = Reference.ById(Guid.Parse(source?.element?.id)),
+                    Value = (source?.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
+                };
+            }
+            else if (type == typeof(UrlSlugElement))
+            {
+                return new UrlSlugElement
+                {
+                    Element = Reference.ById(Guid.Parse(source?.element?.id)),
+                    Mode = source?.mode?.ToString(),
+                    Value = source?.value?.ToString()
+                };
+            }
+            else if (type == typeof(CustomElement))
+            {
+                return new CustomElement
+                {
+                    Element = Reference.ById(Guid.Parse(source?.element?.id)),
+                    Value = source?.value?.ToString(),
+                    SearchableValue = source?.searchable_value?.ToString()
+                };
+            }
+            else if(type == typeof(SubpagesElement))
+            {
+                return new SubpagesElement
+                {
+                    Element = Reference.ById(Guid.Parse(source?.element?.id)),
+                    Value = (source?.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
+                };
+            }
+
+            throw new ArgumentOutOfRangeException($"{type} is not valid language variant element");
```

Comment:
Please update the exception message 
```suggestion
            throw new ArgumentOutOfRangeException($"{type} is not a valid element");
```

Reply:
Good catch, I did not realize when copy-pasting. Changed.